src/v/iceberg/rest_client/catalog_client.cc

```diff
@@ -123,11 +123,9 @@ ss::future<expected<iobuf>> catalog_client::perform_request(
 
     std::vector<http_call_error> retriable_errors{};
 
-    auto client_ptr = _client_source.get().acquire();
     while (true) {
         const auto permit = rtc.retry();
         if (!permit.is_allowed) {
-            co_await client_ptr->shutdown_and_stop();
```

Comment:
Curious where these are called now? I suppose the catalog client needs its own stop method now that it's full owner of the HTTP client

Reply:
i have a commit introducing shutdown method, in the client public interface, missed adding it to this PR